src/watchdog_fsevents.c

```diff
@@ -773,19 +806,25 @@ static PyObject *
 watchdog_remove_watch(PyObject *self, PyObject *watch)
 {
     UNUSED(self);
-    PyObject *streamref_capsule = PyDict_GetItem(watch_to_stream, watch);
+    PyObject *streamref_capsule = NULL;
+    LOCK();
+    PyDict_GetItemRef(watch_to_stream, watch, &streamref_capsule);
+    UNLOCK();
     if (!streamref_capsule) {
         // A watch might have been removed explicitly before, in which case we can simply early out.
         Py_RETURN_NONE;
     }
+    LOCK();
     PyDict_DelItem(watch_to_stream, watch);
+    UNLOCK();
```

Comment:
Same issue. This is a preexisting problem but this call doesn't check the return value and might incorrectly ignore a Python exception.